wsagent/che-core-api-languageserver/src/main/java/org/eclipse/che/api/languageserver/registry/LanguageServerRegistryImpl.java

```diff
@@ -52,31 +75,47 @@
   private final AtomicInteger serverId = new AtomicInteger();
 
   /** Started {@link LanguageServer} by project. */
-  private final Map<String, List<LanguageServerLauncher>> launchedServers;
+  private final Set<String> launchedServers;
 
-  private final Map<String, List<InitializedLanguageServer>> initializedServers;
+  private final Set<InitializedLanguageServer> initializedServers;
 
-  private final Provider<ProjectManager> projectManagerProvider;
-  private final ServerInitializer initializer;
-  private EventService eventService;
-  private CheLanguageClientFactory clientFactory;
+  private final EventService eventService;
+  private final CheLanguageClientFactory clientFactory;
+
+  private static ClientCapabilities CLIENT_CAPABILITIES = initClientCapabilities();
+  private static final int PROCESS_ID = getProcessId();
+
+  private final List<ServerInitializerObserver> observers = new ArrayList<>();
+
+  private static int getProcessId() {
+    String name = ManagementFactory.getRuntimeMXBean().getName();
```

Comment:
it looks like very plaftorm/OS dependent to grab a process id so I'm not sure it's a good choice but it was there previously. we may add a TODO change as with DefaultProcessManager.java and CLibrary or with Java9 the getPid() is coming as well

Reply:
I agree that this looks strange, but I just move this code, so I'd like to leave it as is in this PR.

Reply:
yes